File: core/src/com/symbol/game/ecs/component/enemy/CorporealComponent.java
package com.symbol.game.ecs.component.enemy;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.utils.Pool;

public class CorporealComponent implements Component, Pool.Poolable {

    public boolean corporeal = true;
    public float incorporealTime = 0f;

    @Override
    public void reset() {
        corporeal = true;
        incorporealTime = 0f;
    }

}

File: core/src/com/symbol/game/ecs/component/enemy/CorporealComponent.kt
package com.symbol.game.ecs.component.enemy

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.utils.Pool

class CorporealComponent : Component, Pool.Poolable {

    var corporeal = true
    var incorporealTime = 0f
    var timer = 0f

    override fun reset() {
        corporeal = true
        incorporealTime = 0f
        timer = 0f
    }

}